tests/test_tomography_gate_set_tomography.py

```diff
@@ -0,0 +1,516 @@
+from qibo.tomography.gate_set_tomography import (
+    GST_execute_circuit,
+    execute_GST,
+    measurement_basis,
+    prepare_states,
+    reset_register,
+)
+
+
+@pytest.mark.parametrize("k", np.arange(0, 4, 1))
+def test_prepare_states_valid_k_single_qubit(k):
+    correct_gates = [[gates.I(0)], [gates.X(0)], [gates.H(0)], [gates.H(0), gates.S(0)]]
+    nqubits = 1
+    circuit = prepare_states(k, nqubits)
+
+    for groundtruth, gate in zip(correct_gates[k], circuit.queue):
+        assert isinstance(gate, type(groundtruth))
+
+
+@pytest.mark.parametrize("k", np.arange(0, 16, 1))
+def test_prepare_states_valid_k_two_qubits(k):
+    correct_gates = [
+        [gates.I(0), gates.I(1)],
+        [gates.I(0), gates.X(1)],
+        [gates.I(0), gates.H(1)],
+        [gates.I(0), gates.H(1), gates.S(1)],
+        [gates.X(0), gates.I(1)],
+        [gates.X(0), gates.X(1)],
+        [gates.X(0), gates.H(1)],
+        [gates.X(0), gates.H(1), gates.S(1)],
+        [gates.H(0), gates.I(1)],
+        [gates.H(0), gates.X(1)],
+        [gates.H(0), gates.H(1)],
+        [gates.H(0), gates.H(1), gates.S(1)],
+        [gates.H(0), gates.S(0), gates.I(1)],
+        [gates.H(0), gates.S(0), gates.X(1)],
+        [gates.H(0), gates.S(0), gates.H(1)],
+        [gates.H(0), gates.S(0), gates.H(1), gates.S(1)],
+    ]
+    nqubits = 2
+    circuit = prepare_states(k, nqubits)
+
+    for groundtruth, gate in zip(correct_gates[k], circuit.queue):
+        assert isinstance(gate, type(groundtruth))
+
+
+@pytest.mark.parametrize("k, nqubits", [(0, 3), (1, 4), (2, 5), (3, 6)])
+def test_prepare_states_valid_k_invalid_nqubits(k, nqubits):
+    # Test for value input with invalid nqubits
+
+    # Check if ValueError is raised
+    with pytest.raises(ValueError, match="nqubits needs to be either 1 or 2"):
+        circuit = prepare_states(k, nqubits)
+
+
+@pytest.mark.parametrize("k, nqubits", [(17, 1), (18, 1), (21, 2), (56, 2)])
+def test_prepare_states_invalid_k_valid_nqubits(k, nqubits):
+    # Check if IndexError is raised
+    with pytest.raises(IndexError):
+        circuit = prepare_states(k, nqubits)
+
+
+# ##################################################################################
+
+
+@pytest.mark.parametrize("j", np.arange(0, 4, 1))
+def test_measurement_basis_value_j_single_qubit(j):
+    # Test for valid input with a single qubit
+    nqubits = 1
+
+    test_circuit = qibo.models.Circuit(nqubits)
+    new_circuit = measurement_basis(j, test_circuit)
+
+    assert isinstance(new_circuit, qibo.models.circuit.Circuit)
+    assert new_circuit.nqubits == nqubits
+
+
+@pytest.mark.parametrize("j", np.arange(0, 16, 1))
+def test_measurement_basis_value_j_two_qubits(j):
+    # Test for valid input with two qubits
+    nqubits = 2
+
+    test_circuit = qibo.models.Circuit(nqubits)
+    new_circuit = measurement_basis(j, test_circuit)
+
+    assert isinstance(new_circuit, qibo.models.circuit.Circuit)
+    assert new_circuit.nqubits == nqubits
```

Comment:
You should try and update these tests as you did for the prepare states. Namely, test the output is the expected one and not only that you get a circuit with the expected number of qubits.